functions/packages.bash

```diff
@@ -141,473 +180,467 @@ homegear_setup() {
   else
     cond_echo "Update... "
-    if ! cond_redirect git -C "$DIRECTORY" pull --quiet origin; then echo "FAILED (git pull)"; return 1; fi
-  fi
-  cond_redirect chown -R "openhab:$username" "$DIRECTORY"
-  cond_redirect chmod -R ug+wX "$DIRECTORY"
-  if ! cond_redirect pip3 install -r "$DIRECTORY"/requirements.txt; then echo "FAILED (requirements)"; return 1; fi
-  cond_redirect cp "$DIRECTORY"/template.service /etc/systemd/system/miflora.service
-  cond_redirect systemctl daemon-reload
-  systemctl start miflora.service || true
-  cond_redirect systemctl status miflora.service
-  if cond_redirect systemctl enable miflora.service; then echo "OK"; else echo "FAILED"; return 1; fi
-  if grep -q "dtoverlay=pi3-miniuart-bt" /boot/config.txt; then
+    if ! cond_redirect git -C "$mifloraDir" pull --quiet origin; then echo "FAILED (git pull)"; return 1; fi
+  fi
+  if ! {
+    cond_redirect chown -R "openhab:${username:-openhabian}" "$mifloraDir"
+    cond_redirect chmod -R ug+wX "$mifloraDir"
+  }; then echo "FAILED (permissons)"; fi
+  cond_echo "Installing required python packages"
+  if ! cond_redirect pip3 install -r "$mifloraDir"/requirements.txt; then echo "OK"; else echo "FAILED (python packages)"; return 1; fi
+
+  echo -n "$(timestamp) [openHABian] Setting up miflora-mqtt-daemon service... "
+  if ! (cp "$mifloraDir"/template.service /etc/systemd/system/miflora.service); then echo "FAILED (copy service)"; fi
+  if ! cond_redirect systemctl enable miflora.service; then echo "FAILED (enable service)"; return 1; fi
+  if cond_redirect systemctl restart miflora.service; then echo "OK"; else echo "FAILED (restart service)"; return 1; fi
+
+  if grep -q "^[[:space:]]*dtoverlay=pi3-miniuart-bt" /boot/config.txt; then
     cond_echo "Warning! The internal RPi3 Bluetooth module is disabled on your system. You need to enable it before the daemon may use it."
   fi
+
   if [ -n "$INTERACTIVE" ]; then
-    if [ $FAILED -eq 0 ]; then
-      whiptail --title "Operation Successful!" --msgbox "$successtext" 15 80
-    else
-      whiptail --title "Operation Failed!" --msgbox "$failtext" 10 60
-    fi
+    whiptail --title "Operation Successful!" --msgbox "$successtext" 16 80
   fi
 }
 
 nginx_setup() {
-  introtext="This will enable you to access the openHAB interface through the normal HTTP/HTTPS ports and optionally secure it with username/password and/or an SSL certificate."
-  failtext="Sadly there was a problem setting up the selected option. Please report this problem in the openHAB community forum or as a openHABian GitHub issue."
-
-  echo "$(timestamp) [openHABian] Setting up Nginx as reverse proxy with authentication... "
-  if [ -n "$INTERACTIVE" ]; then
-    if ! (whiptail --title "Description, Continue?" --yes-button "Continue" --no-button "Back" --yesno "$introtext" 15 80); then echo "CANCELED"; return 0; fi
+  if [ -z "$INTERACTIVE" ]; then
```

Comment:
same question: test installed ?

Reply:
Not yet, I am working my way through the tests of the functions but in theory it should work (not necessarily in practice :).